src/Console/Output/Progress/ProgressOutputFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Console\Output\Progress;
+
+use PhpCsFixer\Console\Output\OutputContext;
+
+/**
+ * @internal
+ */
+final class ProgressOutputFactory
+{
+    public function create(string $outputType, OutputContext $context): ProgressOutputInterface
+    {
+        if (null === $context->getOutput()) {
+            $outputType = ProgressOutputType::NONE;
+        }
+
+        if (!self::isBuiltInType($outputType)) {
+            throw new \RuntimeException(
+                sprintf(
+                    'Something went wrong, "%s" output type is not supported',
+                    $outputType
+                )
+            );
+        }
+
+        return ProgressOutputType::NONE === $outputType
```

Comment:
```suggestion
        /** @TODO: PHP 8.0+ - convert to `match` */
        return ProgressOutputType::NONE === $outputType
```

Reply:
There's no point in adding such comment here because it will be refactored anyway in consequent PR #6842.

Reply:
in generla - for reviewing and ultimately aiming for merge of one PR, I do not assume what may or may not be changed in other PR that may or may not get merged 🤷🏻

Reply:
Even without that PR, converting this to `match` does not make much sense with only 2 possible paths - ternary is good enough 😅.

Reply:
match is more extendable ;P